src/main/java/com/spikes2212/command/genericsubsystem/commands/smartmotorcontrollersubsystem/MoveSmartMotorControllerSubsystem.java

```diff
@@ -0,0 +1,155 @@
+package com.spikes2212.command.genericsubsystem.commands.smartmotorcontrollersubsystem;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.revrobotics.CANSparkMax;
+import com.spikes2212.command.genericsubsystem.smartmotorcontrollersubsystem.SmartMotorControllerSubsystem;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+import java.util.function.Supplier;
+
+/**
+ * A command that moves a {@link SmartMotorControllerSubsystem} using its master motor controller's control loops.
+ *
+ * @author Yoel Perman Brilliant
+ * @see SmartMotorControllerSubsystem
+ */
+public class MoveSmartMotorControllerSubsystem extends CommandBase {
+
+    /**
+     * The {@link SmartMotorControllerSubsystem} this command will run on.
+     */
+    protected final SmartMotorControllerSubsystem subsystem;
+
+    /**
+     * The setpoint this command should bring the {@link SmartMotorControllerSubsystem} to.
+     */
+    protected final Supplier<Double> setpoint;
```

Comment:
order the fields in the same order as the constructor parameters (this should also be added to the conventions)